pkg/zval/zval.go

```diff
@@ -71,7 +71,7 @@ func AppendValue(dst []byte, val []byte) []byte {
 // the zval in val.
 func sizeBytes(val []byte) int {
 	// This really is correct even when data is nil.
-	return sizeUvarint(1+uint64(len(val))) + len(val)
+	return sizeUvarint((uint64(len(val)))<<1) + len(val)
```

Comment:
I think a more robust approach is to use newTag:
```suggestion
	return sizeUvarint(newTag(false, len(val)) + len(val)
```